lib/dictator/plug/authorize.ex

```diff
@@ -3,20 +3,12 @@ defmodule Dictator.Plug.Authorize do
 
   @behaviour Plug
 
-  @default_actions [:index, :show, :new, :create, :edit, :update, :delete]
-
   @impl Plug
   def init(opts), do: opts
 
   @impl Plug
   def call(conn, opts) do
-    allowed_actions = Keyword.get(opts, :only, @default_actions)
-
-    if conn.private.phoenix_action in allowed_actions do
-      authorize(conn, opts)
-    else
-      conn
-    end
+    authorize(conn, opts)
```

Comment:
I'm torn on this. I agree with your arguments but I think there's value in allowing users to customise what actions they'd want to validate against.

The problem is that this is specific to Phoenix and I don't think there's an easy way to get this information (please do correct me if I'm wrong). The way I see it we have one of two options (assuming we don't find any other way to get the information):

1. We scrap this option
2. We assume the Phoenix dependency for this option and document it accordingly.

I'm not particularly fond on 2 because it's a pretty big detail to include in a note in the docs.

Reply:
> I think there's value in allowing users to customise what actions they'd want to validate against

I agree with this. But that would be done with a `only:` or `except:` optional argument. Not by assuming a default whitelist.

So my suggestion would be to just remove the current behaviour, and implement that optional one as a separate feature


> The problem is that this is specific to Phoenix

This also forces this plug to be called after the router. I currently cannot add a policy to an entire scope. I have to add it to every individual controller inside it.
Documenting this and stating that `only:`/`except:` would only work after the router would be an acceptable tradeoff, in my opinion

Reply:
I'm a bit confused, what option are you advocating here? Documenting `only:`/`except:` or implementing it an optional, separate feature?

Reply:
What I'm advocating is:

1. remove the current whitelist (aka: what this PR implements).
2. afterwards, implement `only/except` as an optional argument to implement whitelisting/blacklisting, and keeping the default behaviour as it is from this PR (aka: run for all actions by default)

Reply:
Ok, I'll be approving and expecting a follow-up to implement both `only` and `except` behaviours.